tests/test_item.py

```diff
@@ -0,0 +1,42 @@
+import os
+import unittest
+import json
+from tempfile import TemporaryDirectory
+from datetime import datetime
+import dateutil
+
+from pystac import *
+from pystac.utils import is_absolute_href
+from tests.utils import (TestCases, RANDOM_GEOM, RANDOM_BBOX)
+
+
+class ItemTest(unittest.TestCase):
+    def test_to_from_dict(self):
+        m = TestCases.get_path('data-files/itemcollections/sample-item-collection.json')
+        with open(m) as f:
+            item_dict_list = json.load(f)['features']
+        
+        item = Item.from_dict(item_dict_list[0])
+        self.assertIsInstance(item, Item)
+        self.assertIsInstance(item.id, str)
```

Comment:
I'd say, these type tests are less valuable since the types are defined in the class ... it's probably going to be a rare case that one of these fails due to an actual bug, and more likely that a refactor would happen that this would be a nuisance test for. 

For me, this is an instance where the fact that the Item to_dict/from_dict gets run as part of the unit test is the value part, as any bugs will normally throw an exception. So it's really a coverage thing - which is helpful for dynamic languages like python.

Instead of an exhaustive list of type checks, I'd recommend picking out a few valuable cases to test values against - e.g. the `get_self_href` below is a good one. These will test the actual workings of the mechanism on this particular test case, rather than the mechanism itself broadly (the latter being more like the testing e.g. of Link types in the link property is more of that sort of test).

Potentially an easier way to do a more full test that is more broadly applicable is to just run the dict through `from_dict` and `to_dict`, and check that the dicts are equal. This might even reveal ways in which that process makes legitimate changes to the input dict.

This could also serve as a broader utility method, so that each of the STACObject subtypes could have a to/from dict test against sample cases.

Reply:
yeah, this all makes sense. I changed the tests to check to to/from methods directly using a function like you mentioned. I put it in `test_utils.py` as a standalone function but let me know if you think it should be a method that goes elsewhere.

I added a few additional checks to test the Item object before it returns a dict but most of that stuff is taken out. Let me know if you want me to add it back in.